source/common/router/config_impl.cc

```diff
@@ -450,7 +450,7 @@ bool VirtualHostImpl::usesRuntime() const {
 VirtualHostImpl::VirtualClusterEntry::VirtualClusterEntry(
     const envoy::api::v2::VirtualCluster& virtual_cluster) {
   if (virtual_cluster.method() != envoy::api::v2::RequestMethod::METHOD_UNSPECIFIED) {
-    method_ = envoy::api::v2::RequestMethod_Name(virtual_cluster.method());
+    method_.value(envoy::api::v2::RequestMethod_Name(virtual_cluster.method()));
```

Comment:
Right, this is a bit beyond my standard C++ foo but I think this has to have been working. My preference here is actually to implement operator= in optional since it's less surprising. If you feel like doing that (and adding trivial test) that would be cool.